File: bibliotheque-backend/src/main/java/com/ibizabroker/bibliotheque/entity/Reservation.java
package com.ibizabroker.bibliotheque.entity;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.Data;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.util.Date;

@Data
@Entity
@EntityListeners(AuditingEntityListener.class)
@Table(name = "Reservation")
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer reservationId;

    @ManyToOne
    @JoinColumn(name = "book_id", nullable = false)
    Books livre;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    Users adherent;

    @Temporal(TemporalType.TIMESTAMP)
    @JsonSerialize(using = JsonDataSerializer.class)
    Date dateReservation;

    @Temporal(TemporalType.TIMESTAMP)
    @JsonSerialize(using = JsonDataSerializer.class)
    Date dateExpiration;

    @Enumerated(EnumType.STRING)
    ReservationStatus statut;
}

File: bibliotheque-frontend/src/app/_model/reservation.ts
export class Reservation {
    reservationId: number;
    livreId: number;
    livreNom: string;
    adherentId: number;
    adherentNom: string;
    dateReservation: Date | string;
    dateExpiration: Date | string;
    statut: string;
}

export class ReservationRequest {
    livreId: number | null = null;
    adherentId: number | null = null;
}
